docs/DeveloperGuide.adoc

```diff
@@ -718,30 +718,175 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
+|4 |`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
 
-|`* *` |user |hide link:#private-contact-detail[private contact details] by default |minimize chance of someone else seeing them by accident
+|5 |`* * *` |user |find my friends through their initials |find my friend’s details easily
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
-|=======================================================================
+|6 |`* * *` |user |undo and redo my commands |undo and redo my recent errors
 
-{More to be added}
+|7 |`* * *` |user |edit my addressBook |make changes to the addressBook
+
+|8 |`* * *` |user |view my contact list in alphabetical order |scroll through the list easily
+
+|9 |`* * *` |user |use alias for commands |use the commands conveniently
+
+|10 |`* * *` |user |find person using tags |find persons based on the different categories
+
+|11 |`* * *` |user |save my person's DOB |track persons' birthdays
+
+|12 |`* * *` |user |find persons by birthday month |find who is having birthdays
+
+|13 |`* * *` |user |add persons with minimal details |have a choice on the amount of details I want to include about a person
+
+|14 |`* * *` |user |view the total number of persons  |know how many persons in AddressBook
+
+|15 |`* * *` |user |pin contacts at the top |find my favourite contacts more easily
+
+|16 |`* *` |user |sort my list in many different ways |organise according to my preference
+
+|17 |`* *` |user |have an AddressBook that can auto capitalise the name |Add new person with more convenience
+
+|18 |`* *` |user |be able to have a choice of viewing less details on persons |view a cleaner list
+
+|19 |`* *` |user |be able to view more details about persons only after selection |view more information about that person with a cleaner list
+
+|20 |`* *` |user |be able to sync my AddressBook with social media |import persons’ details
+
+|21 |`* *` |user |view selected person’s social media |view more information
+
+|22 |`* *` |user |see invalid commands with a different colour |so that I can identify them easily
+
+|23 |`* *` |user |find persons using emails |find them without knowing their names
+
+|24 |`* *` |user |view selected person’s address visually |have a better idea of where they are at
+
+|25 |`* *` |user |close the help UI with a command |close the help UI without going back to it while on the main window
+
+|26 | `* *` |user |delete persons that contain a certain tag |delete a group with convenience
+
+|27 |`* *` |user |have confirmation before deleting a person |ensure that I am deleting the correct person
+
+|28 |`* *` |user |hide link:#private-contact-detail[private contact details] by default |minimize chance of someone else seeing them by accident
+
+|29 |`*` |user |open one help UI at any time |limit the number of windows opened
```

Comment:
While it is good to have such in-depth, detailed user stories, you might want to narrow down the list of user stories such that it pertains to the main functionalities of the product and not nitpicking on a specific functionality of your product.